demos/coreMQTT/mqtt_demo_plaintext.c

```diff
@@ -0,0 +1,969 @@
+        {
+            xMQTTStatus = MQTT_ProcessLoop( &xMQTTContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
+
+            if( xMQTTStatus != MQTTSuccess )
+            {
+                lReturnStatus = EXIT_FAILURE;
+            }
+        }
+
+        /**************************** Disconnect. *****************************/
+
+        if( lReturnStatus == EXIT_SUCCESS )
+        {
+            /* Send an MQTT Disconnect packet over the already connected TCP
+             * socket. There is no corresponding response for the disconnect
+             * packet. After sending a disconnect, the client must close the
+             * network connection. */
+            LogInfo( ( "Disconnecting the MQTT connection with %s.\r\n", democonfigMQTT_BROKER_ENDPOINT ) );
+            xMQTTStatus = MQTT_Disconnect( &xMQTTContext );
+
+            if( xMQTTStatus != MQTTSuccess )
+            {
+                lReturnStatus = EXIT_FAILURE;
+            }
+        }
+
+        if( lReturnStatus == EXIT_SUCCESS )
+        {
+            /* Close the network connection.  */
+            xNetworkStatus = SecureSocketsTransport_Disconnect( &xNetworkContext );
+
+            if( xNetworkStatus != TRANSPORT_SOCKET_STATUS_SUCCESS )
+            {
+                lReturnStatus = EXIT_FAILURE;
+                LogError( ( "SecureSocketsTransport_Disconnect() failed to close "
+                            "the network connection with status code %d.",
+                            ( int ) xNetworkStatus ) );
+            }
+            else
+            {
+                xIsConnectionEstablished = pdFALSE;
+            }
+        }
+
+        if( lReturnStatus == EXIT_SUCCESS )
+        {
+            /* Reset SUBACK status for each topic filter after completion of
+             * subscription request cycle. */
+            for( ulTopicCount = 0; ulTopicCount < mqttexampleTOPIC_COUNT; ulTopicCount++ )
+            {
+                xTopicFilterContext[ ulTopicCount ].xSubAckStatus = MQTTSubAckFailure;
+            }
+
+            /* Wait for some time between two iterations to ensure that we do not
+             * bombard the broker. */
+            LogInfo( ( "RunCoreMqttPlaintextDemo() completed an iteration successfully. "
+                       "Total free heap is %u.\r\n",
+                       xPortGetFreeHeapSize() ) );
+            LogInfo( ( "Short delay before starting the next iteration.... \r\n\r\n" ) );
+            vTaskDelay( mqttexampleDELAY_BETWEEN_DEMO_ITERATIONS );
+        }
+        else
+        {
+            LogInfo( ( "RunCoreMqttPlaintextDemo() failed an iteration. "
+                       "Total free heap is %u.\r\n",
+                       xPortGetFreeHeapSize() ) );
+            break;
+        }
+    }
+
+    /* An error may have occurred after a successful network connection, so we
+     * close the network connection that may be taking up system resources. */
+    if( xIsConnectionEstablished == pdTRUE )
```

Comment:
This logic can be avoided if the MQTT and transport disconnect calls within the loop (lines 481-497) be updated to use the `  if(( lReturnStatus == EXIT_SUCCESS )  || (xIsConnectionEstablished == pdTRUE ))` condition

Reply:
sure